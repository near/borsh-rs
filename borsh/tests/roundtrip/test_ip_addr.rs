use std::net::{IpAddr, Ipv4Addr, Ipv6Addr};

#[test]
fn test_ipv4_addr_roundtrip_enum() {
    let original = IpAddr::V4(Ipv4Addr::new(192, 168, 0, 1));
    let encoded = borsh::to_vec(&original).expect("Serialization failed");
    #[cfg(feature = "std")]
    insta::assert_debug_snapshot!(encoded);
    let decoded = borsh::from_slice::<IpAddr>(&encoded).expect("Deserialization failed");
    assert_eq!(original, decoded);
}

#[test]
fn test_ipv4_addr_roundtrip() {
    let original = Ipv4Addr::new(192, 168, 0, 1);
    let encoded = borsh::to_vec(&original).expect("Serialization failed");
    #[cfg(feature = "std")]
    insta::assert_debug_snapshot!(encoded);
    let decoded = borsh::from_slice::<Ipv4Addr>(&encoded).expect("Deserialization failed");
    assert_eq!(original, decoded);
}

#[test]
fn test_ipv6_addr_roundtrip_enum() {
    let original = IpAddr::V6(Ipv6Addr::new(0x2001, 0xdb8, 0, 0, 0, 0, 0, 1));
    let encoded = borsh::to_vec(&original).expect("Serialization failed");
    #[cfg(feature = "std")]
    insta::assert_debug_snapshot!(encoded);
    let decoded = borsh::from_slice::<IpAddr>(&encoded).expect("Deserialization failed");
    assert_eq!(original, decoded);
}

#[test]
fn test_ipv6_addr_roundtrip() {
    let original = Ipv6Addr::new(0x2001, 0xdb8, 0, 0, 0, 0, 0, 1);
    let encoded = borsh::to_vec(&original).expect("Serialization failed");
    #[cfg(feature = "std")]
    insta::assert_debug_snapshot!(encoded);
    let decoded = borsh::from_slice::<Ipv6Addr>(&encoded).expect("Deserialization failed");
    assert_eq!(original, decoded);
}

#[test]
fn test_ipaddr_vec_roundtrip() {
    let original = vec![
        IpAddr::V4(Ipv4Addr::new(192, 168, 0, 1)),
        IpAddr::V6(Ipv6Addr::new(0x2001, 0xdb8, 0, 0, 0, 0, 0, 1)),
        IpAddr::V6(Ipv6Addr::new(0xfe80, 0, 0, 0, 0, 0, 0, 1)),
    ];
    let encoded = borsh::to_vec(&original).expect("Serialization failed");
    #[cfg(feature = "std")]
    insta::assert_debug_snapshot!(encoded);
    let decoded = borsh::from_slice::<Vec<IpAddr>>(&encoded).expect("Deserialization failed");
    assert_eq!(original, decoded);
}
