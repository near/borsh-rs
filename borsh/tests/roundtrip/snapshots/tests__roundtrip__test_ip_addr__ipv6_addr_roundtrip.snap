---
source: borsh/tests/roundtrip/test_ip_addr.rs
expression: encoded
---
[
    32,
    1,
    13,
    184,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    1,
]
